RestrictedSamples/WidgetForegroundWorkerSampleCS/Widget1.xaml.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices.WindowsRuntime;
+using System.Threading;
+using System.Threading.Tasks;
+using Windows.Foundation;
+using Windows.Foundation.Collections;
+using Windows.Storage;
+using Windows.Storage.Pickers;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Controls.Primitives;
+using Windows.UI.Xaml.Data;
+using Windows.UI.Xaml.Input;
+using Windows.UI.Xaml.Media;
+using Windows.UI.Xaml.Navigation;
+using Microsoft.Gaming.XboxGameBar;
+using Microsoft.Gaming.XboxGameBar.Restricted;
+using Windows.UI.Core;
+
+// The Blank Page item template is documented at https://go.microsoft.com/fwlink/?LinkId=234238
+
+namespace WidgetForegroundWorkerSample
+{
+    /// <summary>
+    /// An empty page that can be used on its own or navigated to within a Frame.
+    /// </summary>
+    public sealed partial class Widget1 : Page
+    {
+        private XboxGameBarWidget widget = null;
+
+        public Widget1()
+        {
+            this.InitializeComponent();
+        }
+
+        protected override void OnNavigatedTo(NavigationEventArgs e)
+        {
+            widget = e.Parameter as XboxGameBarWidget;
+        }
+
+        private void LaunchForegroundWorkButton_Click(object sender, RoutedEventArgs e)
+        {
+            _ = Task.Run(async () =>

```

Comment:
Would this be simplified by just making the LaunchForegroundWorkButton_Click method async? 
private async void LaunchForegroundWorkButton_Click

Reply:
yeah, I suppose I could. In general async void in c# is a bad idea. Exceptions seem to be made for event handlers, however: https://stackoverflow.com/questions/19415646/should-i-avoid-async-void-event-handlers
I had forgotten about this, so I will change it since it's ok.